lib/validators/modelValidator.ts

```diff
@@ -784,9 +784,14 @@ export class ModelValidator extends SpecValidator<SpecValidationResult> {
             foundIssues = true
             break
           }
-          // replacing forward slashes with empty string because this messes up Sways regex
+
+          // replacing characters that may cause validator failed  with empty string because this messes up Sways regex
           // validation of path segment.
-          parameterValue = parameterValue.replace(/\//gi, "")
+          if (!utils.isUrlEncoded(parameterValue as string)) {
+            // TODO: we can get the scheme from parameterValue if the useSchemePrefix is setting false in the x-ms-parameterized-host,
+            // then check if it can match to the swagger scheme.
+            parameterValue = parameterValue.replace(/[^0-9a-zA-Z._]/gi, "")
```

Comment:
Can we just replace `:` and `/` since there're other valid characters? such as `()&?` .

Reply:
Due to the result of  test , these characters are also invalid : ()&?
Since I only remain  the characters that are valid in urlencode except *+- , can ensure the value is valid,other

Reply:
Let's use whitelist as is now since the value is not validated furthermore.